### ISSUE TITLE
@DataPoints Boolean fails if null is first in array

### ISSUE THREAD

**Issue by reecefowell** [junit-team/junit4#1322]
The following compiles and runs successfully:

```Java
    @DataPoints
    public static Boolean[] criteriaEnabled() {
        return new Boolean[] {
            true,
            false,
            null
        };
    }
```

The following does not:

```Java
    @DataPoints
    public static Boolean[] criteriaEnabled() {
        return new Boolean[] {
            null,
            true,
            false
        };
    }
```

Producing the following error:

```
java.lang.AssertionError: Never found parameters that satisfied method assumptions.  Violated assumptions: []

	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.experimental.theories.Theories$TheoryAnchor.evaluate(Theories.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

It is when NULL is first in the array of booleans we get the error, but not the last in the array.

**Comment by reecefowell** [comment]
Incidently, when NULL is last, although the code runs, a NULL Boolean is never considered as a test case data point that can be injected.

**Comment by reecefowell** [comment]
I see that this was resolved in 4.12, I was using 4.11, so I am closing this ticket now.
